### PR TITLE
Add Sigmoid Function

### PR DESCRIPTION
**Pull request by naitoh**
Add Sigmoid Function
Add Activation (ReluTest/LogSoftmaxTest/SigmoidTest) Test

### CODE REVIEW

File: lib/chainer/functions/activation/log_softmax.rb [hunk 1]
```diff
@@ -10,26 +10,78 @@ def self.logsumexp(x)
...
           retain_outputs([0])
-          x[0][x[0]<=0] = 0
-          [Utils::Array.force_array(x[0])] 
+          [y]
         end
 
-        def backward_cpu(x, gy)
-          y = output_data[0]
-          [Utils::Array.force_array(gy[0] * (y > 0))]
+        def backward(x, gy)
```
Comment (hatappi) [inline lib/chainer/functions/activation/log_softmax.rb]: Why did you rename backward_cpu to backward?

File: lib/chainer/functions/activation/sigmoid.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        #       [0.119203, 0.5, 0.880797]
+        #
+        def self.sigmoid(x)
+          self.new.(x)
+        end
+
+        def forward_cpu(x)
+          half = 0.5
+          y = Utils::Array.force_array((Numo::NMath.tanh(x[0] * half) * half)+ half)
+          retain_outputs([0])
```
Comment (hatappi) [inline lib/chainer/functions/activation/sigmoid.rb]: I think you probably need `retain_inputs([])`.

File: lib/chainer/functions/activation/log_softmax.rb (hunk 1 above)
Comment (naitoh) [reply lib/chainer/functions/activation/log_softmax.rb]: Since the original Chainer source code "chainer/functions/activation/log_softmax.py" is described by CPU & GPU common code with method name of forward() & backward(), it was adopted as it was.

Is it better to stay as it is?

File: lib/chainer/functions/activation/sigmoid.rb (hunk 2 above)
Comment (naitoh) [reply lib/chainer/functions/activation/sigmoid.rb]: Thank you for pointing out.
I agree.
I will fix it.

File: lib/chainer/functions/activation/log_softmax.rb (hunk 1 above)
Comment (hatappi) [reply lib/chainer/functions/activation/log_softmax.rb]: https://github.com/chainer/chainer/blob/v2.0.2/chainer/functions/activation/log_softmax.py
I was misunderstanding.
A common `forward`, `backward` function was used for cpu and gpu.

**Comment by hatappi** [comment]
Thank you for contributing 🎉
